db/compaction/compaction_outputs.cc

```diff
@@ -489,15 +489,28 @@ Status CompactionOutputs::AddRangeDels(
     if (upper_bound != nullptr) {
       int cmp =
           ucmp->CompareWithoutTimestamp(*upper_bound, tombstone.start_key_);
-      if ((has_overlapping_endpoints && cmp < 0) ||
-          (!has_overlapping_endpoints && cmp <= 0)) {
-        // Tombstones starting after upper_bound only need to be included in
-        // the next table. If the current SST ends before upper_bound, i.e.,
-        // `has_overlapping_endpoints == false`, we can also skip over range
-        // tombstones that start exactly at upper_bound. Such range
-        // tombstones will be included in the next file and are not relevant
-        // to the point keys or endpoints of the current file.
+      // Tombstones starting after upper_bound only need to be included in
```

Comment:
I am confused about why in some places the range tombstone's scope starts from kMaxSequenceNumber on the start key (`FragmentedRangeTombstoneIterator::parsed_start_key()`), while in other places (`tombstone_start` in this function) the range tombstone's scope starts from the range tombstone seqno. Is there any logic behind this or it's all legacy inconsistencies?

Reply:
I may be able to ignore this inconsistency for this review, under the assumption that, if they were to converge, we would make `FragmentedRangeTombstoneIterator::parsed_start_key()` use the range tombstone seqno for the start of the range tombstone's scope. Then we would have the same problem no matter which starting scope is used.